pkg/container-hook/tracer.go

```diff
@@ -257,12 +257,13 @@ func (n *ContainerNotifier) installEbpf(fanotifyFd int) error {
 		return fmt.Errorf("reading private data from fanotify fd: %w", err)
 	}
 
-	consts := map[string]interface{}{
-		"tracer_group": fanotifyPrivateData,
+	execSpec := &execruntimeSpecs{}
+	if err := spec.Assign(&execSpec); err != nil {
+		return err
 	}
-	//nolint:staticcheck
-	if err := spec.RewriteConstants(consts); err != nil {
-		return fmt.Errorf("RewriteConstants: %w", err)
+
+	if err := execSpec.TracerGroup.Set(fanotifyPrivateData); err != nil {
```

Comment:
`execruntimeSpecs` is just a structure containing the specs, it doesn't provide methods like LoadAndAssign. TBH I'm not sure how it's designed in that way, IMO it would make more sense for `loadExecruntime` to return `execruntimeSpecs` instead of `*ebpf.CollectionSpec`, but there is probably a strong good reason not to follow that approach.